plugins/com.google.cloud.tools.eclipse.appengine.deploy.test/src/com/google/cloud/tools/eclipse/appengine/deploy/DeployPreferencesTest.java

```diff
@@ -16,60 +16,151 @@
   public void testDefaultVersion() {
-    assertThat(defaultPreferences.getVersion(), isEmptyString());
+    assertThat(DeployPreferences.DEFAULT_CUSTOM_VERSION, isEmptyString());
+    assertThat(preferences.getVersion(), isEmptyString());
+  }
+
+  @Test
+  public void testSetVersion() {
+    assertThat(preferences.getVersion(), isEmptyString());
+    preferences.setVersion("someversion97");
+    assertThat(preferences.getVersion(), is("someversion97"));
+    preferences.setVersion(null);
+    assertThat(preferences.getVersion(), isEmptyString());
   }
 
   @Test
   public void testDefaultAutoPromote() {
-    assertTrue(defaultPreferences.isAutoPromote());
+    assertTrue(DeployPreferences.DEFAULT_ENABLE_AUTO_PROMOTE);
+    assertTrue(preferences.isAutoPromote());
+  }
+
+  @Test
+  public void testSetAutoPromote() {
+    assertTrue(preferences.isAutoPromote());
+    preferences.setAutoPromote(false);
+    assertFalse(preferences.isAutoPromote());
+    preferences.setAutoPromote(true);
+    assertTrue(preferences.isAutoPromote());
   }
 
   @Test
   public void testDefaultBucket() {
-    assertThat(defaultPreferences.getBucket(), isEmptyString());
+    assertThat(DeployPreferences.DEFAULT_CUSTOM_BUCKET, isEmptyString());
+    assertThat(preferences.getBucket(), isEmptyString());
+  }
+
+  @Test
+  public void testSetBucket() {
+    assertThat(preferences.getBucket(), isEmptyString());
+    preferences.setBucket("somebucket45");
+    assertThat(preferences.getBucket(), is("somebucket45"));
+    preferences.setBucket(null);
+    assertThat(preferences.getBucket(), isEmptyString());
   }
 
   @Test
   public void testDefaultStopPreviousVersion() {
-    assertTrue(defaultPreferences.isStopPreviousVersion());
+    assertTrue(DeployPreferences.DEFAULT_STOP_PREVIOUS_VERSION);
+    assertTrue(preferences.isStopPreviousVersion());
+  }
+
+  @Test
+  public void testSetStopPreviousVersion() {
+    assertTrue(preferences.isStopPreviousVersion());
+    preferences.setStopPreviousVersion(false);
+    assertFalse(preferences.isStopPreviousVersion());
+    preferences.setStopPreviousVersion(true);
+    assertTrue(preferences.isStopPreviousVersion());
+  }
+
+  @Test
+  public void testDefaultIncludeOptionalConfigurationFiles() {
+    assertTrue(DeployPreferences.DEFAULT_INCLUDE_OPTIONAL_CONFIGURATION_FILES);
+    assertTrue(preferences.isIncludeOptionalConfigurationFiles());
   }
 
   @Test
-  public void testIncludeOptionalConfigurationFiles() {
-    assertTrue(defaultPreferences.isIncludeOptionalConfigurationFiles());
+  public void testSetIncludeOptionalConfigurationFiles() {
+    assertTrue(preferences.isIncludeOptionalConfigurationFiles());
+    preferences.setIncludeOptionalConfigurationFiles(false);
+    assertFalse(preferences.isIncludeOptionalConfigurationFiles());
+    preferences.setIncludeOptionalConfigurationFiles(true);
+    assertTrue(preferences.isIncludeOptionalConfigurationFiles());
   }
 
+  @Test
+  public void testResetToDefault() {
```

Comment:
It doesn't imply `save()`, because `resetToDefaults()` is for when you press the _Reset Defaults_ button, and that just resets the intermediate model.

> If no, `DeployPreferences` could record a value to indicate that save() should remove the node.

Sorry, I didn't get it. Why remove the preference node?